crates/bevy_ecs/src/world/mod.rs

```diff
@@ -592,64 +592,106 @@ impl World {
         }
     }
 
+    /// Inserts a new resource with standard starting values.
+    ///
+    /// If the resource already exists, nothing happens.
+    ///
+    /// The value given by the [`FromWorld::from_world`] method will be used.
+    /// Note that any resource with the `Default` trait automatically implements `FromWorld`,
+    /// and those default values will be here instead.
+    #[inline]
+    pub fn init_resource<R: Resource + FromWorld>(&mut self) {
+        // PERF: We could avoid double hashing here, since the `from_resources` call is guaranteed
```

Comment:
```suggestion
        // PERF: We could avoid double hashing here, since the `from_world` call is guaranteed
```

Reply:
The corrections are certainly less wrong than the existing comments though :p